### PR TITLE
Feature/web3modal/connect wallet

### PR DESCRIPTION
**Pull request by kacperoak**
- Add ability to connect new wallet via web3Modal
- Improvements on UI 

connect

scan_qr

help

### CODE REVIEW

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/ui/routes/composable_routes/chain_selection/ChainSelectionRoute.kt [hunk 1]
```diff
@@ -10,36 +11,56 @@ import androidx.compose.foundation.lazy.itemsIndexed
...
+        }
+    }
+
     ChainSelectionScreen(
         chains = chainsState,
         onChainClick = viewModel::updateChainSelectState,
         onConnectClick = {
-            // Navigate to web3Modal for now
-            navController.navigateToWeb3Modal()
+            if (viewModel.isAnyChainSelected) {
```
Comment (kacperoak) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/ui/routes/composable_routes/chain_selection/ChainSelectionRoute.kt:57]: Paring selection from settled parings will be added in next PR

**Review by Elyniss — APPROVED** [review]
LTGM! Awesome work!

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/ui/routes/composable_routes/chain_selection/ChainSelectionRoute.kt [hunk 2]
```diff
@@ -10,36 +11,56 @@ import androidx.compose.foundation.lazy.itemsIndexed
...
 
 @Composable
 fun ChainSelectionRoute(navController: NavController) {
+    val context = LocalContext.current
     val viewModel: ChainSelectionViewModel = viewModel()
     val chainsState by viewModel.uiState.collectAsState()
 
+    LaunchedEffect(Unit) {
+        viewModel.walletEvents.collect {
+            when(it) {
```
Comment (TalhaAli00) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/ui/routes/composable_routes/chain_selection/ChainSelectionRoute.kt:46]: can you replace `it` with a variable name?

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/DappSampleApp.kt [hunk 3]
```diff
@@ -45,5 +47,9 @@ class DappSampleApp : Application() {
         PushDappClient.initialize(Push.Dapp.Params.Init(CoreClient, null)) { error ->
             Timber.e(tag(this), error.throwable.stackTraceToString())
         }
+
+        startKoin {
+            modules(web3ModalModule())
+        }
```
Comment (TalhaAli00) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/DappSampleApp.kt:53]: will this be removed once we move W3M into it's own module?

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3Modal.kt [hunk 4]
```diff
@@ -31,16 +48,34 @@ fun Web3Modal(
...
+    val web3ModalViewModel: Web3ModalViewModel = viewModel()
     val navController = rememberAnimatedNavController()
     val coroutinesScope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    LaunchedEffect(Unit) {
+        web3ModalViewModel
+            .walletEvents
+            .onEach {
+                when(it) {
```
Comment (TalhaAli00) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3Modal.kt:60]: same comment about replacing `it`

**Review by TalhaAli00 — APPROVED** [review]

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/DappSampleApp.kt (hunk 3 above)
Comment (kacperoak) [reply samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/DappSampleApp.kt:53]: Yes

**Pull request merged**